### PR TITLE
Logistic regression solution notebook refactoring

### PR DESCRIPTION
**Pull request by carlotta94c**
- Fixing features encoding (using ordinal encoder for Item size, categorical encoder for all the other features and label encoder for Color) as recommended by scikit-learn documentation 
-  Using F1 score instead of accuracy to evaluate the model since the dataset is highly skewed
- Updating plots to better show the data before and after encofings and the relationships between label and features 
- Fixing ROC curve plot using matplotlib

### CODE REVIEW

**Comment by bstollnitz** [comment]
Looks great!

**Review by bstollnitz — APPROVED** [review]

**Comment by jimbobbennett** [comment]
Should we remove the output from this notebook? Would make it smaller and easier to diff

**Comment by gdequeiroz** [comment]
> Should we remove the output from this notebook? Would make it smaller and easier to diff

@jimbobbennett 
Do you mean just for the PR or in general?  Since this is a solution notebook, we would like to have the output (+ we are trying to be consistent with the other solution notebooks from the other sections.

**Comment by gdequeiroz** [comment]
> > > Should we remove the output from this notebook? Would make it smaller and easier to diff
> > 
> > 
> > @jimbobbennett Do you mean just for the PR or in general? Since this is a solution notebook, we would like to have the output (+ we are trying to be consistent with the other solution notebooks from the other sections.
> 
> I was thinking in general, just this notebook was in the PR so made me think of it. Is it intentional to have the output in the notebooks then?

Yes!  :)

**Comment by jimbobbennett** [comment]
Can't add this inline as the file is too large, but in the Build your model section, this line:

```python
from sklearn.model_selection import train_test_split
```

is in both of the first 2 cells, it doesn't need to be in the second one.

image
